ming_sim/action_materialize.py

```
@@ -844,6 +846,225 @@ def _materialize_grant_allocation(ctx: MaterializeCtx) -> None:
+        if absolute_end > 0:
+            staged["end_turn"] = absolute_end
+        parsed_ongoing = _parse_json_field(ongoing_effects)
+        if isinstance(parsed_ongoing, dict) and parsed_ongoing:
+            staged["ongoing_effects"] = parsed_ongoing
+    if existing_id:
+        return db.update_directive_candidate(existing_id, staged)
+    return db.stage_directive_candidate(int(turn), minister_name, payload=staged)
+
+
+def _assignment_dossier_text(ctx: MaterializeCtx) -> str:
+    """案卷 text：ADR 0028 最近相关对话上下文链 + 本轮皇帝/大臣句。
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)